core/dbt/config/renderer.py

```diff
@@ -55,12 +55,49 @@ def render_data(
             )
 
 
+def _list_if_none(value):
+    if value is None:
+        value = []
+    return value
+
+
+def _dict_if_none(value):
+    if value is None:
+        value = {}
+    return value
+
+
+def _list_if_none_or_string(value):
+    value = _list_if_none(value)
+    if isinstance(value, str):
+        return [value]
+    return value
+
+
+class ProjectPostprocessor(Dict[Keypath, Callable[[Any], Any]]):
```

Comment:
This logic used to live in the Project itself, but it's really part of rendering. This also avoids us having to make a second pass of `deep_map` through the config.